src/GraphQL/Validation/TypeInfo.cs

```diff
@@ -5,6 +5,9 @@
 
 namespace GraphQL.Validation
 {
+    /// <summary>
+    /// Provides information pertaining to the current state of the AST tree while being walked.
```

Comment:
This is one of the most confusing classes for me.

Reply:
Yup, but I expect that it saves a lot of allocations and code.